plugins/inputs/execd/execd.go

```diff
@@ -112,9 +119,7 @@ func (e *Execd) cmdLoop(ctx context.Context) error {
 		case <-ctx.Done():
 			if e.stdin != nil {
 				e.stdin.Close()
-				// Immediately exit process but with a graceful shutdown
-				// period before killing
-				internal.WaitTimeout(e.cmd, 200*time.Millisecond)
+				internal.GracefulStop(e.cmd, 200*time.Millisecond)
```

Comment:
Now I'm off topic, but 200ms seems too short.  I think we should give 5s to shutdown before kill.

Reply:
I'd be open to increasing it, but I wonder what effect this will have on the restart delay or other things; if they'll wait in line properly or launch another process without waiting.

Reply:
It should only slow down restart if the process takes longer than this to exit when sent a SIGTERM.  FWIW, Debian has a 30s delay in start_stop_daemon while systemd defaults to 90s.